MkChrootTree.sh

```diff
@@ -99,7 +100,16 @@ else
    for (( IDX=${#PARTS[@]}-1 ; IDX>=0 ; IDX-- ))
    do
       # Get partition-label
-      LABEL=$(e2label "/dev/${PARTS[IDX]}")
+      if [[ ${DEVFSTYP} == ext3 ]] ||
+         [[ ${DEVFSTYP} == ext4 ]]
+      then
+         LABEL=$(e2label "/dev/${PARTS[IDX]}")
+      elif [[ ${DEVFSTYP} == xfs ]]
+      then
+         LABEL=$( xfs_admin -l "/dev/${PARTS[IDX]}" | sed -e 's/"$//' -e 's/^.*"//' )
+      else
+         err_out 1 "Unable to determine fstype of /dev/${PARTS[IDX]}"
```

Comment:
This error message doesn't appear to track with the if/else, since `DEVFSTYP` is not compared to `/dev/${PARTS[IDX]}`. What am I missing?

Reply:
Yeah. It's a bit baroque. I forget the specifics of why I had to resort to this ("mists of time" and all that). There was something peculiar about setting up the chroot for creating a "recovery" AMI that led me down this, particular thorny-path.

At any rate, the `DEVFSTYP` is used for telling the script how to read the filesystem label from the disk-partition identified by `/dev/${PARTS[IDX]}`. What I was comparing _might_ have been more obvious in a `case` structure than an `if`/`elif`/`else structure.

Reply:
It's really just the wording of the message that is throwing me. It says it is unable to determine the fstype, but it looks more to me that, as constructed, the user specified an unsupported `${DEVFSTYP}`. I think the current message would make more sense if `${PARTS[IDX]}` were used in the if/elif conditions, in a comparison to `${DEVFSTYP}` or xfs|ext3|ext4.

But, the message is distinct enough that search will turn up the code, so, /shrug.